src/scaffold.js

```diff
@@ -0,0 +1,193 @@
+
+// package.json
+
+const generatePackageJson = ({ subgraphName }) =>
+  prettier.format(
+    JSON.stringify({
+      name: getSubgraphBasename(subgraphName),
+      license: 'UNLICENSED',
+      scripts: {
+        codegen: 'graph codegen',
+        build: 'graph build',
+        deploy:
+          `graph deploy ` +
+          `--node https://api.thegraph.com/deploy/ ` +
+          `--ipfs https://api.thegraph.com/ipfs/ ` +
+          subgraphName,
+        'create-local': `graph create --node http://localhost:8020/ ${subgraphName}`,
+        'remove-local': `graph remove --node http://localhost:8020/ ${subgraphName}`,
+        'deploy-local':
+          `graph deploy ` +
+          `--node http://localhost:8020/ ` +
+          `--ipfs http://localhost:5001 ` +
+          subgraphName,
+      },
+      dependencies: {
+        '@graphprotocol/graph-cli': '0.7.2',
+        '@graphprotocol/graph-ts': '0.5.1',
+      },
+    }),
+    { parser: 'json' }
+  )
+
+// Subgraph manifest
+
+const eventSignature = event =>
+  `${event.name}(${event.inputs.map(input => input.type).join(',')})`
+
+const generateManifest = ({ abi, address, network, subgraphName }) =>
+  prettier.format(
+    `
+specVersion: 0.0.1
+schema:
+  file: ./schema.graphql
+dataSources:
+  - kind: ethereum/contract
+    name: Contract
+    network: ${network}
+    source:
+      address: '${address}'
+      abi: Contract
+    mapping:
+      kind: ethereum/events
+      apiVersion: 0.0.1
+      language: wasm/assemblyscript
+      entities:
+        ${abiEvents(abi)
+          .map(event => `- ${event.name}`)
+          .join('\n        ')}
+      abis:
+        - name: Contract
+          file: ./abis/Contract.json
+      eventHandlers:
+        ${abiEvents(abi)
+          .map(
+            event => `
+        - event: ${eventSignature(event)}
+          handler: handle${event.name}`
+          )
+          .join('')}
+      file: ./src/mapping.ts
+`,
+    { parser: 'yaml' }
+  )
+
+// Schema
+
+const ethereumTypeToGraphQL = name => {
+  let ascType = ascTypeForEthereum(name)
+  return valueTypeForAsc(ascType)
+}
+
+const generateEventType = event => `type ${event.name} @entity {
+      id: ID!
+      ${event.inputs
+        // TODO: Support inputs without names
+        .map(
+          (input, index) =>
+            `${input.name || `param${index}`}: ${ethereumTypeToGraphQL(input.type)}! # ${
+              input.type
+            }`
+        )
+        .join('\n')}
+    }`
+
+const generateSchema = ({ abi }) =>
+  prettier.format(
+    abiEvents(abi)
+      .map(generateEventType)
+      .join('\n'),
+    {
+      parser: 'graphql',
+    }
+  )
+
+// Mapping
+
+const generateMapping = ({ abi, subgraphName }) =>
+  prettier.format(
+    `
+  import { ${abiEvents(abi).map(
+    event => `${event.name} as ${event.name}Event`
+  )}} from '../generated/Contract/Contract'
+  import { ${abiEvents(abi).map(event => event.name)} } from '../generated/schema'
```

Comment:
This assumes that generated code lives in `/generated`, which would be a good name, but currently codegen puts things in `/types`.

Reply:
Not on master. I switched to `generated` as part of the Gluegun port, although now I feel bad because it's a breaking change. But the generated code here will work.